data-access-services/transactions-service/src/main/java/at/deloittedigital/core_banking/transactions/service/TransactionService.java

```diff
@@ -13,11 +13,14 @@
     private final TransactionRepository transactionRepository;
 
     /**
-     * Retrieves all transactions.
+     * Retrieves transactions by account ID. Returns all transactions if no account ID is specified.
      *
-     * @return all transactions
+     * @return transactions by account ID or all transactions
      */
-    public List<Transaction> getAllTransactions() {
-        return transactionRepository.findAll();
+    public List<Transaction> getTransactions(Long accountId) {
```

Comment:
Suggestion: `@Nullable` annotation for the parameter so if anyone tries to use it w/o null check, static code analysis will flag it.